src/cpp/ext/otel/otel_plugin.cc

```diff
@@ -527,14 +535,38 @@ OpenTelemetryPlugin::OpenTelemetryPlugin(
                                              descriptor.instrument_type));
         }
         for (size_t i = 0; i < descriptor.optional_label_keys.size(); ++i) {
-          if (optional_label_keys->find(descriptor.optional_label_keys[i]) !=
-              optional_label_keys->end()) {
+          if (optional_label_keys.find(descriptor.optional_label_keys[i]) !=
+              optional_label_keys.end()) {
             instruments_data_[descriptor.index].optional_labels_bits.set(i);
           }
         }
       });
 }
 
+namespace {
+constexpr absl::string_view kLocality = "grpc.lb.locality";
+}
+
+absl::string_view OpenTelemetryPlugin::OptionalLabelKeyToString(
+    grpc_core::ClientCallTracer::CallAttemptTracer::OptionalLabelKey key) {
+  switch (key) {
+    case grpc_core::ClientCallTracer::CallAttemptTracer::OptionalLabelKey::
+        kLocality:
+      return kLocality;
+    default:
+      grpc_core::Crash("Illegal OptionalLabelKey index");
+  }
```

Comment:
Since we only have one valid OptionalLabelKey that the user can set for now, we could also write it like:

```
  GPR_ASSERT(key ==
      grpc_core::ClientCallTracer::CallAttemptTracer::OptionalLabelKey::kLocality);
  return kLocality;
```

Reply:
The current structure makes it easier to add new labels, and I think the compiler would be smart enough to optimize it.